app/src/features/surveys/observations/sampling-sites/SamplingSiteList.tsx

```diff
@@ -9,173 +9,210 @@ import { grey } from '@mui/material/colors';
 import IconButton from '@mui/material/IconButton';
 import List from '@mui/material/List';
 import ListItem from '@mui/material/ListItem';
+import ListItemIcon from '@mui/material/ListItemIcon';
 import ListItemText from '@mui/material/ListItemText';
+import Menu, { MenuProps } from '@mui/material/Menu';
+import MenuItem from '@mui/material/MenuItem';
 import Toolbar from '@mui/material/Toolbar';
 import Typography from '@mui/material/Typography';
+import { CodesContext } from 'contexts/codesContext';
+import { SurveyContext } from 'contexts/surveyContext';
+import { useContext, useEffect, useState } from 'react';
 import { useHistory } from 'react-router';
+import { getCodesName } from 'utils/Utils';
 
 const SamplingSiteList = () => {
   const history = useHistory();
 
+  const surveyContext = useContext(SurveyContext);
+  const codesContext = useContext(CodesContext);
+
+  useEffect(() => {
+    codesContext.codesDataLoader.load();
+  }, [codesContext.codesDataLoader]);
+
+  surveyContext.sampleSiteDataLoader.load(surveyContext.projectId, surveyContext.surveyId);
+
+  const [anchorEl, setAnchorEl] = useState<MenuProps['anchorEl']>(null);
+  const [selectedSampleSiteId, setSelectedSampleSiteId] = useState<number | undefined>();
+
+  const handleMenuClick = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>, sample_site_id: number) => {
+    setAnchorEl(event.currentTarget);
+    setSelectedSampleSiteId(sample_site_id);
+  };
+
+  if (
+    !surveyContext.sampleSiteDataLoader.data ||
+    (surveyContext.sampleSiteDataLoader.isLoading && !codesContext.codesDataLoader.data) ||
+    codesContext.codesDataLoader.isLoading
+  ) {
+    // TODO add loading animation
+    return <></>;
```

Comment:
There was one before, but it was just rendering in the top left corner. I can throw it back for now, but it will need updating regardless to make it render nicely

Reply:
Added